VultisigApp/VultisigApp/Chains/cosmos/cosmos.swift

```diff
@@ -0,0 +1,246 @@
+//
+//  cosmos.swift
+//  VultisigApp
+//
+//  Created by Enrique Souza Soares on 19/11/24.
+//
+
+import Foundation
+import WalletCore
+import Tss
+import CryptoSwift
+
+class CosmosHelper {
+    var coinType: CoinType
+    var denom: String
+    var gasLimit: UInt64
+    
+    init(coinType:CoinType, denom: String, gasLimit: UInt64){
+        self.coinType = coinType
+        self.denom = denom
+        self.gasLimit = gasLimit
+    }
+    
+    func getSwapPreSignedInputData(keysignPayload: KeysignPayload, signingInput: CosmosSigningInput) throws -> Data {
+            guard case .Cosmos(let accountNumber, let sequence,let gas, _, let ibc) = keysignPayload.chainSpecific else {
+                throw HelperError.runtimeError("fail to get account number and sequence")
+            }
+            guard let pubKeyData = Data(hexString: keysignPayload.coin.hexPublicKey) else {
+                throw HelperError.runtimeError("invalid hex public key")
+            }
+            var input = signingInput
+            input.publicKey = pubKeyData
+            input.accountNumber = accountNumber
+            input.sequence = sequence
+            input.mode = .sync
+            
+            input.fee = CosmosFee.with {
+                $0.gas = self.gasLimit
+                $0.amounts = [CosmosAmount.with {
+                    $0.denom = self.denom
+                    $0.amount = String(gas)
+                }]
+            }
+            // memo has been set
+            // deposit message has been set
+            return try input.serializedData()
+        }
+        
+    
+    func getPreSignedInputData(keysignPayload: KeysignPayload) throws -> Data {
+        guard case .Cosmos(let accountNumber, let sequence , let gas, _, let ibc) = keysignPayload.chainSpecific else {
+            throw HelperError.runtimeError("getPreSignedInputData: fail to get account number and sequence")
+        }
+        guard let pubKeyData = Data(hexString: keysignPayload.coin.hexPublicKey) else {
+            throw HelperError.runtimeError("getPreSignedInputData: invalid hex public key")
+        }
+        let coin = self.coinType
+        
+        if keysignPayload.coin.isNativeToken
```

Comment:
for this one , like on Terra , UST , we can't set it as native token , and it is not IBC token either ,this will fail , correct?

Reply:
nevermind , terra is in a seperate one